packages/core/http/core-http-browser/src/types.ts

```diff
@@ -130,6 +143,25 @@ export interface IExternalUrl {
   validateUrl(relativeOrAbsoluteUrl: string): URL | null;
 }
 
+/**
+ * APIs for creating hrefs to static assets.
+ *
+ * @public
+ */
+export interface IStaticAssets {
+  /**
+   * Gets the full href to the current plugin's asset,
+   * given its path relative to the plugin's `public/assets` folder.
+   *
+   * @example
+   * ```ts
+   * // I want to retrieve the href for the asset stored under `my_plugin/public/assets/some_folder/asset.png`:
+   * const assetHref = core.http.statisAssets.getPluginAssetHref('some_folder/asset.png');
+   * ```
+   */
+  getPluginAssetHref(assetPath: string): string;
+}
```

Comment:
I'm only exposing an API to retrieve plugin-scoped assets right now (meaning a plugin can't link to assets from another plugin). I suspect we may need to add another API for it later, but I wanted to try keeping the asset isolation for now.